pyproject.toml

```diff
@@ -136,23 +136,34 @@ ignore = [
     # minimal cost for standard lib imports; keep this disabled
     "TCH003", # typing-only-standard-library-import
     # gives false positives if we use try imports and type-checking import
-    "TCH004", # runtime-import-in-type-checking-block
     "TID252", # Relative imports from parent modules are banned
     # https://beta.ruff.rs/docs/rules/#flake8-use-pathlib-pth
     "PTH123", # pathlib-open - this would force pathlib usage anytime open or with open was used.
 ]
+# https://docs.astral.sh/ruff/linter/#fix-safety
+extend-safe-fixes = [
+    # pydantic models use annotations at runtime and need to be accounted for
+    # these are safe to apply automatically given settings for tool.ruff.flake8-type-checking
+    "TCH", # https://beta.ruff.rs/docs/rules/#flake8-type-checking-tch
+]
```

Comment:
There's a lot of changes in this PR (applied by ruff) but these `ruff` config changes are the important ones.